OpenRA.Mods.Common/Traits/EmitInfantryOnSell.cs

```diff
@@ -23,27 +23,44 @@ class EmitInfantryOnSellInfo : TraitInfo<EmitInfantryOnSell>
 		[ActorReference]
 		[Desc("Be sure to use lowercase. Default value is \"e1\".")]
 		public readonly string[] ActorTypes = { "e1" };
+
+		[Desc("Emits actors only if this field contains the owner's Race.")]
+		public readonly string[] Races = { };
+
+		public object Create(ActorInitializer init) { return new EmitInfantryOnSell(init.Self, this); }
 	}
 
-	class EmitInfantryOnSell : INotifySold
+	public class EmitInfantryOnSell : INotifySold
 	{
+		public readonly EmitInfantryOnSellInfo Info;
+
+		readonly bool correctRace = false;
+
+		public EmitInfantryOnSell(Actor self, EmitInfantryOnSellInfo info)
+		{
+			Info = info;
+			correctRace = info.Races.Contains(self.Owner.Country.Race);
```

Comment:
This should also be set to true if `Races` is empty.

Either that, or you need an upgrade rule, otherwise you'll probably break a bunch of maps and mods.